packages/flame/lib/src/game/notifying_vector2.dart

```diff
@@ -0,0 +1,190 @@
+import 'dart:typed_data';
+import 'package:flutter/foundation.dart';
+import 'package:vector_math/vector_math_64.dart';
+
+/// Extension of the standard [Vector2] class, implementing the [ChangeNotifier]
+/// functionality. This allows any interested party to be notified when the
+/// value of this vector changes.
+///
+/// This class can be used as a regular [Vector2] class. However, if you do
+/// subscribe to notifications, don't forget to eventually unsubscribe in
+/// order to avoid resource leaks.
+///
+/// Direct access to this vector's [storage] is not allowed.
+///
+// ignore: prefer_mixin
+class NotifyingVector2 extends Vector2 with ChangeNotifier {
```

Comment:
some places we don't allow for you to set your own instance, eg PositionComponent.position IIRC has some custom setters. I wonder what scenarios this would not work with

Reply:
Isn't that exactly the case that this will be used for? So that #904 can recalculate the matrix for example

Reply:
Yes, this is how it is being used by `Transform2D`: it exposes `.position` as a `NotifyingVector2`, and subscribes to any changes in that vector. Thus, when the user says, for example, `transform.position.setFrom(v)` the `transform` gets notified about the fact that its position has changed.

Reply:
ah, I see. I thought this was more for users of Flame to use on their components, but rather the idea is to use it ourselves on PositionComponent. gotcha.